eclipse-langium/langium
# Add type hierarchy provider as Language Server capability

Closes #1275

* Handle type hierarchy requests in the Language Server
* Add `TypeHierarchyProvider` as a new LSP service
* Add the interface `TypeHierarchyProvider` and a no-op abstract implementation `AbstractTypeHierarchyProvider` to easily customize the responses for a given language
* Add a type hierarchy provider for interfaces in Langium grammars

**Test instructions:**

Some automated tests are included. To check functionality in VS Code, create a Langium grammar with the following content:

```ts
interface A {}
interface B extends A {}
```

Place the cursor on the name of either interface and invoke the action "Types: Show Type Hierarchy". Via the button 
![image](https://github.com/eclipse-langium/langium/assets/2501322/93610573-4236-49bb-973d-b8c46f2ab58d)
you can switch between showing subtypes and supertypes of the selected interface.
